lib/rubocop/cop/internal_affairs/example_heredoc_delimiter.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module InternalAffairs
+      # Use `RUBY` for heredoc delimiter of example Ruby code.
+      #
+      # Some editors may apply better syntax highlighting by using appropriate language names for
+      # the delimiter.
+      #
+      # @safety
+      #   This cop is unsafe because the heredoc body may also contain a `RUBY` line,
+      #   in which case this word cannot be used for the delimiter.
```

Comment:
Since this is detectable, it should be able to be made safe. Can you add tests and adjust the implementation? An approach could be that this case is only detected and not auto-corrected.

Reply:
That's right. Looking at some existing codes, there are cases where we really want to use a `RUBY` line in heredoc body, so I decided not to make such a case an offense.